llvm/include/llvm/Support/Parallel.h

```diff
@@ -30,14 +30,6 @@ namespace parallel {
 extern ThreadPoolStrategy strategy;
 
 #if LLVM_ENABLE_THREADS
-#define GET_THREAD_INDEX_IMPL                                                  \
```

Comment:
The check for (threadIndex != UINT_MAX) seems useful(done by https://reviews.llvm.org/D148916). It ensures that thread_index is not mistakenly clashed with main thread.  I vote for keeping it and removing only "sequential" part.

Suggestion done by @aganea allows to keep this check - https://github.com/llvm/llvm-project/pull/109084#discussion_r1765635073